STM32/AC/Core/Src/ACBoard.c

```diff
@@ -182,22 +184,39 @@ static void CAallOn(bool isOn)
 }
 static void CAportState(int port, bool state, int percent, int duration)
 {
+	uint8_t pwmPercent;
+	getPWMPinPercent(port-1, &pwmPercent);
+	// If heat sink has reached the maximum allowed temperature and user
+	// tries to heat the system further up then disregard the input command
+	if (heatSinkTemperature > MAX_TEMPERATURE && percent > pwmPercent)
+		return;
+
     actuatePins((ActuationInfo) { port - 1, percent, 1000*duration, true });
 }
 
 static void heatSinkLoop()
 {
+	static unsigned long previous = 0;
     // Turn on fan if temp > 55 and turn of when temp < 50.
     if (stmGetGpio(fanCtrl) && isFanAutoOn)
     {
         if (heatSinkTemperature < 50)
         {
             stmSetGpio(fanCtrl, false);
         }
-        else if (heatSinkTemperature > 55)
-        {
-            stmSetGpio(fanCtrl, true);
-        }
+    }
+    else if (heatSinkTemperature > 55)
```

Comment:
should this also be checking isFanAutoOn?

also I suggest to get rid of the nested if above, although that might require explicitely checking the getgpio result in both if